### PR TITLE
Vikram's homework 8

### PR DESCRIPTION
**Pull request by ghvik**

### CODE REVIEW

File: HW08_ch11_ex02c.py [hunk 1]
```diff
@@ -20,18 +20,46 @@ def reverse_lookup_old(d, v):
...
+            for word in line.split():
+                word = "".join(letter for letter in word if letter not in punctuation)
+                pledge_list.append(word)
+    return pledge_list
 
 ###############################################################################
 # INSERT COMPLETED CODE FROM HW08_ch11_ex02a BELOW: ###########################
 ###############################################################################
 def main():   # DO NOT CHANGE BELOW
+    pledge_histogram = histogram_new(get_pledge_list())
```
Comment (OhRobin) [inline HW08_ch11_ex02c.py]: thx. got some help here

File: mimsmind1.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        num_digits = int(sys.argv[1])
+    else:
+        num_digits = 3
+    secret_number = randint(0, 10**num_digits - 1)
+    secret_number = to_0_padded_string(secret_number, num_digits)
+    guesses = 0
+    maxrounds = (2**num_digits) + num_digits
+    guessed_correctly = False
+    for n in range(maxrounds):
+        user_input = input(("Please guess a {}-digit number. You have {} tries. "
```
Comment (danielsgriffin) [inline mimsmind1.py]: Notice the msg and the input prompt in the example are on the same line.

File: mimsmind1.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+#
+# Mims Place-in Exam 1
+# Mastermind
+#
+# usage: python mimsmind1.py [length]
+##############################################################################
+
+from random import randint
```
Comment (danielsgriffin) [inline mimsmind1.py]: (1) Please have your print statements match exactly.
(2) Please check your cow counts.

Example:
```
Let's play the mimsmind1 game. You have 11 guesses.

Guess a 3-digit number: 789

0 bull(s), 0 cow(s). Try again: 891

0 bull(s), 1 cow(s). Try again: 189

1 bull(s), 0 cow(s). Try again: 135

1 bull(s), 1 cow(s). Try again: 111

1 bull(s), 0 cow(s). Try again: 132

1 bull(s), 2 cow(s). Try again: 125

2 bull(s), 0 cow(s). Try again: 123

Congratulations. You guessed the correct number in 8 tries.
```
You
```
Please guess a 3-digit number. You have 11 tries. 789
You have guessed 1 time(s)
bull(s): 0, cow(s): 0. Try again.
Please guess a 3-digit number. You have 11 tries. 891
You have guessed 2 time(s)
bull(s): 0, cow(s): 1. Try again.
Please guess a 3-digit number. You have 11 tries. 189
You have guessed 3 time(s)
bull(s): 1, cow(s): 1. Try again.
Please guess a 3-digit number. You have 11 tries. 135
You have guessed 4 time(s)
bull(s): 1, cow(s): 2. Try again.
Please guess a 3-digit number. You have 11 tries. 111
You have guessed 5 time(s)
bull(s): 1, cow(s): 1. Try again.
Please guess a 3-digit number. You have 11 tries. 132
You have guessed 6 time(s)
bull(s): 1, cow(s): 3. Try again.
Please guess a 3-digit number. You have 11 tries. 125
You have guessed 7 time(s)
bull(s): 2, cow(s): 2. Try again.
Please guess a 3-digit number. You have 11 tries. 123
You have guessed 8 time(s)
Congratulations! You guessed the correct number in 8 tries!
```

**Comment by danielsgriffin** [comment]
Check your print statements and your cow counts. Good job.

File: mimsmind1.py (hunk 3 above)
Comment (ghvik) [inline mimsmind1.py]: I think I have it correct in the homework 9 submission.. Forgot to call that the final commit.

**Pull request closed without merging**